### PR TITLE
Add Skybox plugin

### LINKED ISSUE #3941: Request: Skyboxes
Would it be possible to add a toggle that allows a skybox? Not asking for anything crazy, a simple shade of light blue would work for me.

Thanks,

### PR DESCRIPTION
**Pull request by Abextm**
Closes #3941
sky1
sky2
sky3

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/callback/Hooks.java [hunk 1]
```diff
@@ -149,6 +151,8 @@ public void clientMainLoop()
 			client.setTickCount(tick + 1);
 		}
 
+		eventBus.post(BEFORE_RENDER);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: Maybe this should be a 'main loop' event? Also should possibly be in a separate commit

Comment (Abextm) [reply runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: I feel like 'main loop` could be confused with a client tick event

File: runelite-client/src/main/java/net/runelite/client/plugins/skybox/SkyboxPlugin.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	{
+		cx -= client.getBaseX() / 8;
+		cy -= client.getBaseY() / 8;
+		try
+		{
+			return client.getInstanceTemplateChunks()[plane][cx][cy];
+		}
+		catch (ArrayIndexOutOfBoundsException e)
+		{
+			return -1;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/skybox/SkyboxPlugin.java]: When does this happen? Generally I avoid having exceptions in the normal control flow. The eventbus should log it anyway.

**Comment by devLotto** [comment]
Can we have a sand color skybox while in the desert? #c2b280

File: runelite-client/src/main/java/net/runelite/client/plugins/skybox/SkyboxPlugin.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+			px = p.getX();
+			py = p.getY();
+		}
+
+		// Inverse of camera location / 2
+		int spx = px - ((client.getCameraX() - px) >> 1);
+		int spy = py - ((client.getCameraY() - py) >> 1);
+
+		client.setSkyboxColor(skybox.getColorForPoint(
+			client.getBaseX() + (spx / 128.f),
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/skybox/SkyboxPlugin.java]: Should this be off of the player location? Otherwise the sky changes if you move the camera which is weird

Comment (devLotto) [reply runelite-client/src/main/java/net/runelite/client/plugins/skybox/SkyboxPlugin.java]: The sky currently changes when moving about using the Oculus orb, which I find nice.
